Failures associated with `dev`

* Failures associated with the `dev` CI environments
* in Python 3.14 dev – #279 
   * [see here](https://github.com/uscuni/neatnet/actions/runs/20144536297/job/57821159748?pr=279#step:6:720)
* in Python 3.13 dev – current CI
  *  [see here](https://github.com/uscuni/neatnet/actions/runs/19979407925/job/57302800588#step:6:719)


<details closed>
<summary>Python 3.14 dev</summary>

```
FAILED neatnet/tests/test_simplify.py::test_neatify_small[exclusion_mask-1.05-65765.0] - assert 66190.18393245876 ± 6.61902 == 65765.0
  
  comparison failed
  Obtained: 65765.0
  Expected: 66190.18393245876 ± 6.61902
FAILED neatnet/tests/test_simplify.py::test_neatify_small[standard-1.5-64566.0] - assert 66028.35750904656 ± 6.60284 == 64566.0
  
  comparison failed
  Obtained: 64566.0
  Expected: 66028.35750904656 ± 6.60284
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[liege_1656-0.3-2350782] - assert 2397827.2406643797 ± 239.783 == 2350782
  
  comparison failed
  Obtained: 2350782
  Expected: 2397827.2406643797 ± 239.783
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[bucaramanga_4617-0.2-1681011] - assert 1733797.4216781147 ± 173.38 == 1681011
  
  comparison failed
  Obtained: 1681011
  Expected: 1733797.4216781147 ± 173.38
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[douala_809-0.1-2961364] - assert 2981676.499544299 ± 298.168 == 2961364
  
  comparison failed
  Obtained: 2961364
  Expected: 2981676.499544299 ± 298.168
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[slc_4881-0.3-1762456] - assert 1806617.8773647698 ± 180.662 == 1762456
  
  comparison failed
  Obtained: 1762456
  Expected: 1806617.8773647698 ± 180.662
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[aleppo_1133-0.2-4361625] - assert 4649429.0989395585 ± 464.943 == 4361625
  
  comparison failed
  Obtained: 4361625
  Expected: 4649429.0989395585 ± 464.943
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[auckland_869-0.3-1268048] - assert 1290578.7663966361 ± 129.058 == 1268048
  
  comparison failed
  Obtained: 1268048
  Expected: 1290578.7663966361 ± 129.058
======= 8 failed, 174 passed, 1 skipped, 1 warning in 709.12s (0:11:49) ========
```

</details>





<details closed>
<summary>Python 3.13 dev</summary>

```
FAILED neatnet/tests/test_simplify.py::test_neatify_small[exclusion_mask-1.05-65765.0] - assert 66190.18393245876 ± 6.61902 == 65765.0
  
  comparison failed
  Obtained: 65765.0
  Expected: 66190.18393245876 ± 6.61902
FAILED neatnet/tests/test_simplify.py::test_neatify_small[standard-1.5-64566.0] - assert 66028.35750904656 ± 6.60284 == 64566.0
  
  comparison failed
  Obtained: 64566.0
  Expected: 66028.35750904656 ± 6.60284
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[auckland_869-0.3-1268048] - assert 1290578.7663966361 ± 129.058 == 1268048
  
  comparison failed
  Obtained: 1268048
  Expected: 1290578.7663966361 ± 129.058
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[liege_1656-0.3-2350782] - assert 2397827.2406643797 ± 239.783 == 2350782
  
  comparison failed
  Obtained: 2350782
  Expected: 2397827.2406643797 ± 239.783
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[bucaramanga_4617-0.2-1681011] - assert 1733797.4216781147 ± 173.38 == 1681011
  
  comparison failed
  Obtained: 1681011
  Expected: 1733797.4216781147 ± 173.38
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[slc_4881-0.3-1762456] - assert 1806617.8773647698 ± 180.662 == 1762456
  
  comparison failed
  Obtained: 1762456
  Expected: 1806617.8773647698 ± 180.662
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[douala_809-0.1-2961364] - assert 2981676.499544299 ± 298.168 == 2961364
  
  comparison failed
  Obtained: 2961364
  Expected: 2981676.499544299 ± 298.168
FAILED neatnet/tests/test_simplify.py::test_neatify_full_fua[aleppo_1133-0.2-4361625] - assert 4649429.0989395585 ± 464.943 == 4361625
  
  comparison failed
  Obtained: 4361625
  Expected: 4649429.0989395585 ± 464.943
```

</details>

* failing values are identical for both Python versions in `dev`
* this will be due to an upstream update, though need to verify new "truth"

----------------------------------

### Major Deps [^1]

|               | ✅ latest 3.13            | ❌ dev 3.13              | ✅ latest 3.14            | ❌ dev 3.14              |
| ---           | ---                       | ---                       | ---                       | ---                       |
| GEOS          | 3.14.1                    | 3.13.1                    | 3.14.1                    | 3.13.1                    |
| GDAL          | 3.12.0                    | 3.11.4                    | 3.12.0                    | 3.11.4                    |
| PROJ          | 9.7.1                     | 9.5.1                     | 9.7.1                     | 9.5.1                     |
| geopandas     | 1.1.1                     | 1.1.2.dev45+gc301579e0    | 1.1.1                     | 1.1.2.dev45+gc301579e0    |
| numpy         | 2.3.5                     | 2.4.0rc1                  | 2.3.5                     | 2.4.0rc1                  |
| pandas        | 2.3.3                     | 3.0.0rc0                  | 2.3.3                     | 3.0.0rc0                  |
| pyproj        | 3.7.2                     | 3.7.2                     | 3.7.2                     | 3.7.2                     |
| shapely       | 2.1.2                     | 2.1.2                     | 2.1.2                     | 2.1.2                     |
| pyogrio       | 0.12.1                    | 0.12.1                    | 0.12.1                    | 0.12.1                    |
| esda          | 2.8.0                     | 2.8.1.dev5+ga4f4ef609     | 2.8.0                     | 2.8.1.dev5+ga4f4ef609     |
| libpysal      | 4.13.0                    | 4.13.1.dev39+g338aff198   | 4.13.0                    | 4.13.1.dev38+g87ca1c4fb   |
| mapclassify   | 2.10.0                    | ⚠️                        | 2.10.0                    | ⚠️                        |
| momepy        | 0.10.0                    | 0.10.1.dev20+g47f811c31   | 0.10.0                    | 0.10.1.dev20+g47f811c31   |
| networkx      | 3.6.1                     | 3.6.1                     | 3.6.1                     | 3.6.1                     |
| osmnx         | 2.0.7                     | ⚠️                        | 1.0.1                     | ⚠️                        |
| scikit-learn  | 1.8.0                     | 1.8.0                     | 1.8.0                     | 1.8.0                     |
| scipy         | 1.16.3                    | 1.17.0rc1                 | 1.16.3                    | 1.17.0rc1                 |

[^1]: `mapclassify` and `osmnx` are not core deps

--- a/neatnet/artifacts.py
+++ b/neatnet/artifacts.py
@@ -352,7 +352,6 @@ def get_artifacts(
     polys["isoareal_index"] = shape.isoareal_quotient(polys.geometry)
     polys["isoperimetric_quotient"] = shape.isoperimetric_quotient(polys.geometry)
 
-    is_artifact = polys["is_artifact"]
     area = polys["area_sqm"]
     isoareal = polys["isoareal_index"]
     isoperimetric = polys["isoperimetric_quotient"]
@@ -365,7 +364,7 @@ def get_artifacts(
     while True:
         # count number of artifacts to break while loop
         # when no new artifacts are added
-        artifact_count_before = sum(is_artifact)
+        artifact_count_before = sum(polys["is_artifact"])
 
         # polygons that are enclosed by artifacts (at this moment) and
         # that are touching artifacts (at this moment)
@@ -397,7 +396,7 @@ def get_artifacts(
         cond_metric = isoperimetric > isoperimetric_threshold_circles_touching
         polys.loc[cond_geom & cond_area & cond_metric, "is_artifact"] = True
 
-        artifact_count_after = sum(is_artifact)
+        artifact_count_after = sum(polys["is_artifact"])
         if artifact_count_after == artifact_count_before:
             break
 

